src/json.c

```diff
@@ -94,26 +94,35 @@ int FieldSpec_CheckJsonType(FieldType fieldType, JSONType type) {
 static int JSON_getFloat32(RedisJSON json, float *val) {
   double temp;
   int ret = japi->getDouble(json, &temp);
-  *val = (float)temp;
-  return ret;
+  if (REDISMODULE_OK == ret) {
+    *val = (float)temp;
+    return ret;
+  } else {
+    // On RedisJSON<2.0.9, getDouble can't handle integer values.
+    long long tempInt;
+    ret = japi->getInt(json, &tempInt);
+    *val = (float)tempInt;
+    return ret;
```

Comment:
@rafie To reach this code, we need to test using an older RedisJSON version, e.g., RediSearch master and RedisJSON 2.0.8  We can do it just for this specific test, agree?